### PR TITLE
#74 improve validators project update + Task  #79

### PR DESCRIPTION
**Pull request by alesyalis**
+ #79 When creating a project, a candidate is not added (project result is an empty array)

### CODE REVIEW

File: Recrutify/Recrutify.Services/Services/CandidateService.cs [hunk 1]
```diff
@@ -89,29 +89,32 @@ public async Task<CandidateDTO> CreateAsync(CandidateCreateDTO candidateCreateDT
...
                 await _candidateRepository.CreateAsync(candidate);
                 var newCanditate = _mapper.Map<CandidateDTO>(candidate);
                 await _projectService.IncrementCurrentApplicationsCountAsync(projectId);
                 return newCanditate;
             }
 
             var candidateToUpdate = _mapper.Map(candidateCreateDTO, currentCandidate.DeepCopy());
-            var primarySkill = _mapper.Map<CandidatePrimarySkill>(candidateCreateDTO.PrimarySkill);
-            var projectResults = currentCandidate.ProjectResults?.ToList();
+            var currentprojectResults = currentCandidate.ProjectResults?.ToList();
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Services/CandidateService.cs:106]: currentProjectResults

File: Recrutify/Recrutify.Services/Validators/UpdateProjectValidator.cs [hunk 2]
```diff
@@ -1,14 +1,55 @@
...
+                .CustomAsync(ProjectAreExistingAsync);
             RuleFor(p => p.Id)
                 .NotEmpty();
+            RuleFor(p => p.EndDate)
+                .NotNull()
+                .GreaterThanOrEqualTo(DateTime.UtcNow.Date)
+                .WithMessage("Date must be in the today or future!");
+        }
+
+        private async Task ProjectAreExistingAsync(UpdateProjectDTO dto, ValidationContext<UpdateProjectDTO> context, CancellationToken cancellationToken)
```
Comment (elizavetapavliv) [inline Recrutify/Recrutify.Services/Validators/UpdateProjectValidator.cs:34]: You are checking here not existence of project. Rename to UmmutableFieldsAreKeepingUnchanged

Comment (alesyalis) [reply Recrutify/Recrutify.Services/Validators/UpdateProjectValidator.cs]: done

File: Recrutify/Recrutify.Services/Services/CandidateService.cs (hunk 1 above)
Comment (alesyalis) [reply Recrutify/Recrutify.Services/Services/CandidateService.cs]: done

**Review by elizavetapavliv — APPROVED** [review]

**Pull request merged**